### ISSUE TITLE
Support your GPUs for 8+ years, like Nvidia does, including gfx906 GPUs

### ISSUE THREAD

**Issue by EwoutH** [RadeonOpenCompute/ROCm#2308]
I was utterly amazed to read this in the ROCm 5.6 [release notes](https://rocm.docs.amd.com/en/latest/CHANGELOG.html#rocm-5-6-0):

> - AMD Instinct MI50, Radeon Pro VII, and Radeon VII products (collectively referred to as gfx906 GPUs) will be entering the maintenance mode starting Q3 2023. This will be aligned with ROCm 5.7 GA release date.
>   - No new features and performance optimizations will be supported for the gfx906 GPUs beyond ROCm 5.7
>   - Bug fixes / critical security patches will continue to be supported for the gfx906 GPUs till Q2 2024 (End of Maintenance [EOM])(will be aligned with the closest ROCm release)

The Vega 20 GPU (gfx906) is barely 5 years old. Even more, the Radeon VII, Radeon Pro VII and Instinct MI50 are still being sold!

If you want to know why everyone is going with Nvidia, there are a lot of them, but what they do at least well is they support they GPUs.

In the CUDA 12.x release from December 2022, Nvidia also [dropped support](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#cuda-libraries) for some GPUs:

> Support for the following compute capabilities is removed for all libraries:
> - sm_35 (Kepler)
> - sm_37 (Kepler)

Yes, you read that right, _Kepler_ GPUs. These are 2014 products. Not 2019 products like you're dropping.

So, hereby I would you strongly urge to not force us all to Nvidia again, by properly supporting your products.

**Comment by johnnynunez** [comment]
> I was utterly amazed to read this in the ROCm 5.6 [release notes](https://rocm.docs.amd.com/en/latest/CHANGELOG.html#rocm-5-6-0):
> 
> > * AMD Instinct MI50, Radeon Pro VII, and Radeon VII products (collectively referred to as gfx906 GPUs) will be entering the maintenance mode starting Q3 2023. This will be aligned with ROCm 5.7 GA release date.
> >   
> >   * No new features and performance optimizations will be supported for the gfx906 GPUs beyond ROCm 5.7
> >   * Bug fixes / critical security patches will continue to be supported for the gfx906 GPUs till Q2 2024 (End of Maintenance [EOM])(will be aligned with the closest ROCm release)
> 
> The Vega 20 GPU (gfx906) is barely 5 years old. Even more, the Radeon VII, Radeon Pro VII and Instinct MI50 are still being sold!
> 
> If you want to know why everyone is going with Nvidia, there are a lot of them, but what they do at least well is they support they GPUs.
> 
> In the CUDA 12.x release from December 2022, Nvidia also [dropped support](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#cuda-libraries) for some GPUs:
> 
> > Support for the following compute capabilities is removed for all libraries:
> > 
> > * sm_35 (Kepler)
> > * sm_37 (Kepler)
> 
> Yes, you read that right, _Kepler_ GPUs. These are 2014 products. Not 2019 products like you're dropping.
> 
> So, hereby I would you strongly urge to not force us all to Nvidia again, by properly supporting your products.

7900xtx and similars including laptoprs are working now, but not has official support. For the whole community it is already great news, they have finally realized that the whole opensource community (consumer gpus) and PhD students can greatly accelerate the progress of this library. Hopefully they will also give native windows support soon, although it is optional, almost everyone uses linux for AI.

**Comment by xcom169** [comment]
What about Vega56 / GFX9  is it still supported somehow?

**Comment by MelihDarcanxyz** [comment]
While the development of NVK (open source Nvidia drivers) is advancing, Nvidia is supporting their GPUs for at least 8 years, and already have the great (and mature) CUDA, I really can't see why a business or an individual should take risk and go with AMD.

I want to support AMD, I really do. But reasons above are hard to ignore.

**Comment by amayra** [comment]
as an RDNA user and I'm starting to be interested in AI there is no way to go with new AND GPU just because you don't went support old GPU like my RX 5700 XT 

CUDA looks so sexy from here when I'm trying experiment with Stable Diffusion  

I moved from windows and linux just to use hack environment variable "HSA_OVERRIDE_GFX_VERSION=10.3.0" to make Stable Diffusion work

**Comment by samuelpmish** [comment]
As a long time (10+ years) CUDA developer, I just bought a Radeon VII (because it was one of the few cards AMD appeared to actually support) so I could try out developing for AMD hardware, too. It's also a great card in the sense that it's one of the few options on the market (from any vendor) to have good FP64 performance at a reasonable price point.

It's disappointing to see AMD orphaning capable hardware like this so early in its lifetime.

**Comment by KEDI103** [comment]
As Radeon VII user and since 2005 using starting with Ati area bought always AMD but this thing force me why I am still going with AMD which I can't get support and suffer so much...
Nvidia users ridicule us which isn't nice as AMD user. 
AMD AI yt stream looks good but I can't get any benefit for AI but nvidia users crushing me. 
Even funny thing they removed Radeon vii page from AMD website. I was trying to compare my cards with modern cards I noticed one day puff it gone.
Should I regret my support and trust to AMD for years?
heres my problem right now...
https://github.com/RadeonOpenCompute/ROCm/issues/2314

**Comment by keryell** [comment]
I have bought recently 3 Radeon VII for some projects at work (which is... AMD). So this is sad to have them no longer supported soon. :-(
But this depends on what is really behind "entering the maintenance mode" or "Bug fixes will not be back ported to older ROCm releases" from https://rocm.docs.amd.com/en/latest/CHANGELOG.html#os-and-gpu-support-changes. At the end, if everything is *really* open-source in the sense that it can really be compiled everywhere with a unified robust build system and good open-source collaborative spirit at AMD, the open-source community could still do some improvement if it is required.
At the end it would not be that different from the competition where you cannot have anyway super modern features usable on latest GPU, except that if it is really open-source like ROCm, you can still extend the lifetime of some hardware beyond what marketing people care about.

**Comment by stalkerg** [comment]
Also, it seems like AMD team has some confusing intentions because AMDKFD interface in the Linux kernel still supports all cards started from GCN, and effectively only ROCm uses such an interface. (rusicls also thinking about it but not yet)
Because the kernel module/interface can exist only if has user space implementation this is a question for Linus and DRM maintainers.

I am a Vega56 user.

**Comment by stalkerg** [comment]
> At the end, if everything is really open-source in the sense that it can really be compiled everywhere with a unified robust build system and good open-source collaborative spirit at AMD, the open-source community could still do some improvement if it is required.

ROCm is open source but not open development. They do even not read the issue tracker usually.

**Comment by drtpotter** [comment]
Yes, please keep supporting gfx906 on ROCM for another few years at least! I use a Radeon VII for HIP training courses and it has by far the best double precision compute available in a consumer grade GPU. It is environmentally irresponsible to force users to discard hardware that still has plenty of life left!

**Comment by ev-dev** [comment]
+1 gfx906 support

I waited for **_years_** to  see ROCm come to Windows, only to find out that my Radeon VII, which was considered one of the best consumer GPUs only 2 years ago, wouldn't be included, not even in future releases! Please consider extending hardware support 🙏

**Comment by EwoutH** [comment]
> AMD Instinct MI50, Radeon Pro VII, and Radeon VII products (collectively gfx906 GPUs) enters maintenance mode in ROCm 6.0.

Nvidia still supports eight year old 2015 GPUs. AMD is dropping support for four year old 2019 GPUs.

When we finally thought we had a feasible Nvidia alternative, it was too good to be true.

Can’t believe we have to rely on Intel of all companies to offer a sustainable alternative.

**Comment by johnnynunez** [comment]
Because Nvidia not change cuda cores drastically over years. AMD is recent new with cdna and rdna so in my opinion  they are working on these architectures right now

**Comment by johnnynunez** [comment]
Added support for additional GPU architectures.
Navi 3 series: gfx1100, gfx1101, and gfx1102.

**Comment by johnnynunez** [comment]
For me it's working very well.
Use my scripts and build tensorflow and pytorch with your id card.
https://github.com/johnnynunez/rocm_lab

**Comment by JustGitting** [comment]
Not sure if this should be added as another issue or placed here...

As pointed out by @stalkerg, rocm is "open source but not open development.". Where decisions to drop support for hardware is made behind closed doors with no input from normal users or clear reasons why.

It appears there is no technical reason why 5 year old AMD cards cannot be supported for a much longer period. I speculate that the reason for AMD's decision is data-centers and HPC customers don't use cards older then 4-5 years...

As others have pointed out, it is environmentally irresponsible to cut support or usability of a perfectly capable piece of kit.
It also denies many people around the world from learning computer science / ML / programming with AMD GPU's if they are forced to buy expensive hardware. This means students and those in poorer communities miss-out on the opportunities to learn in this area. Which leads people to go (back) to Nvidia as older and cheaper hardware is supported.

I, like most people, hope AMD do the right thing.

Community/AMD, is it possible to (re)add support for formally dropped cards?

From @stalkerg's [comment](https://github.com/ROCm/ROCm/issues/2308#issuecomment-1659518982) the limitation is not in the driver (amdgpu), but within rocm itself (?).

Is there any documentation or information about adding/maintaining support in rocm for unsupported GPUs?

Would AMD be open to accepting PR's from the community to support unofficial/dropped cards? 
Or could support be added in a modular way without direct inclusion in rocm?

**Comment by alann-sapone** [comment]
> as an RDNA user and I'm starting to be interested in AI there is no way to go with new AND GPU just because you don't went support old GPU like my RX 5700 XT
> 
> CUDA looks so sexy from here when I'm trying experiment with Stable Diffusion
> 
> I moved from windows and linux just to use hack environment variable "HSA_OVERRIDE_GFX_VERSION=10.3.0" to make Stable Diffusion work

Hi :)
I've been struggling to make it work and so far I didn't succeed...
Could you provide me any kind of guide (or a docker image) to finally be able to make my 5700XT work ? :)

Thanks =)

**Comment by Na3MnO4** [comment]
Recently bought 2 Radeon Pro VII with Infinity Fabric Link Bridge. GFX906 is definitely not out of date, really couldn't understand this act. Also for the ROCm Windows version, GFX906 and GFX1012 are even not supported by the Runtime. So it is even not possible to run stable diffusion on Windows. I would say it is just threatening "Old" customers by using software disabling.

**Comment by PhilipDeegan** [comment]
I want to buy an Radeon VII pro, the specs are great and the price is great, but this is legitimately insane.

**Comment by SMH17** [comment]
> I was utterly amazed to read this in the ROCm 5.6 [release notes](https://rocm.docs.amd.com/en/latest/CHANGELOG.html#rocm-5-6-0):
> 
> > * AMD Instinct MI50, Radeon Pro VII, and Radeon VII products (collectively referred to as gfx906 GPUs) will be entering the maintenance mode starting Q3 2023. This will be aligned with ROCm 5.7 GA release date.
> >   
> >   * No new features and performance optimizations will be supported for the gfx906 GPUs beyond ROCm 5.7
> >   * Bug fixes / critical security patches will continue to be supported for the gfx906 GPUs till Q2 2024 (End of Maintenance [EOM])(will be aligned with the closest ROCm release)
> 
> The Vega 20 GPU (gfx906) is barely 5 years old. Even more, the Radeon VII, Radeon Pro VII and Instinct MI50 are still being sold!
> 
> If you want to know why everyone is going with Nvidia, there are a lot of them, but what they do at least well is they support they GPUs.
> 
> In the CUDA 12.x release from December 2022, Nvidia also [dropped support](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#cuda-libraries) for some GPUs:
> 
> > Support for the following compute capabilities is removed for all libraries:
> > 
> > * sm_35 (Kepler)
> > * sm_37 (Kepler)
> 
> Yes, you read that right, _Kepler_ GPUs. These are 2014 products. Not 2019 products like you're dropping.
> 
> So, hereby I would you strongly urge to not force us all to Nvidia again, by properly supporting your products.

I totally agree, but I have to correct you about the age of the no more supported Kepler GPUs mentioned for comparison:  actually they are **2012** GPUs!
